README.md

```diff
@@ -17,1529 +17,197 @@
 
 The executor MUST be the UCAN delegate. Their DID MUST be set the in `aud` field of the contained UCAN.
 
-## 2.2 Components
+## Components
 
-### 2.2.1 Task
+### Task
 
 A [Task] is like a deferred function application: a request to perform some action on a resource with specific input.
 
-### 2.2.2 Authorization
-
-An [Authorization] is a cryptographically signed proof permitting execution of referenced [Task]s. It allows the [Invoker] to authorize a group of tasks using one cryptographic signature.
-
-### 2.2.3 Invocation
-
-An [Invocation] is a command to the [Executor] to run the [Task], authorized by the [Invoker].
-
-### 2.2.4 Result
+### Result
 
 A [Result] is the output of a [Task].
 
-### 2.2.5 Receipt
-
-A [Receipt] is a cryptographically signed description of the [Invocation] output and requested [Effect]s.
+### Effect
 
-### 2.2.6 Effect
+An [Effect] is a [Task] caused by the invoked [Task].
 
-An [Effect] are the instruction to the [Executor] to run set of [Task]s concurrently.
-
-## 2.3 IPLD Schema
+## IPLD Schema
 
 ```ipldsch
-type Task struct {
-  on      URI
-  call    Ability
-  input   {String : Any}
+type UCAN union {
+  | Invocation "ucan/i/1.0.0-rc.1"
+  | Receipt    "ucan/o/1.0.0-rc.1"
+} representation keyed
 
-  nnc     string
+type Receipt struct {
+  p SignaturePayload
+  s Signature
 }
 
-type URI string
-type Ability string
-
-type Authorization struct {
-  # Authorization is denoted by the set of links been authorized
-  scope   [&Any]
-  # Scope signed by the invoker
-  s       VarSig
+type SignaturePayload struct {
+  h VarsigHeader
+  p ReceiptPayload
 }
 
-type Invocation struct {
-  v       SemVer
-
-  run     &Task
+type ReceiptPayload union {
+  | AssertPayload "/ucan/assert"
+} representation inline {
+  discriminantKey "cmd"
+}
 
-  # Receipt of the invocation that caused this invocation
-  cause   optional &Invocation
+type AssertPayload struct {
+  uiv SemVer
 
-  # Task authorization.
-  auth    &Authorization
+  iss DID
+  sub DID
+  aud DID
 
-  meta    {String : Any}
+  args Assertion
+  nonce String
 
-  prf     [&UCAN]
+  meta Meta
+  prf [&Delegation]
+  exp Expiry
 }
 
-type SemVer string
-
-type Receipt struct {
-  # Invocation this is a receipt for
-  ran     &Invocation
+type Assertion struct {
+  about &Task
+  facts InvocationFacts
+}
 
+type InvocationFacts struct {
   # Output of the invocation
   out     Result
 
   # Effects to be performed
-  fx      Effects
- 
-  # All the other metadata
-  meta    {String : Any}
-
-  # Principal that issued this receipt. If omitted issuer is
-  # inferred from the invocation task audience.
-  iss     optional Principal
-
-  # When issuer is different from executor this MUST hold a UCAN
-  # delegation chain from executor to the issuer. This should be 
-  # omitted when the executor is the issuer.
-  prf     [&UCAN]
-
-  # Signature from the "iss".
-  s       Varsig
+  run     Effect[]
```

Comment:
@expede your comment about asserts been very generic and my sentiment that receipt is just an assertion of `{out, run}` got me wondering that perhaps `run` could be made optional and implicitly `[]`. That way this is even more minimal requirement space. 

Or alternatively we could make make the whole thing as

```js
{
   cmd: '/ucan/assert'
   args: {
       about: { '/': 'bafy...thing' },
       facts: {
          receipt: {
              out: { ok: {} },
              run: [] 
          }
      }
   }
}
```

That way `/ucan/assert/receipt` could be typed as follows and be equivalent of the above

```js
{
   cmd: '/ucan/assert/receipt'
   args: {
       about: { '/': 'bafy...thing' },
       facts: {
         out: { ok: {} },
         run: [] 
      }
   }
}
```

Reply:
Hmm I'm not sure about adding `facts` as a wrapper around `out` and `run`. Does it add something that I'm missing? This is such a common case, it makes me wonder if we should just put receipts at the same level as assertions ("receipts don't need to be assertions")

Reply:
> Hmm I'm not sure about adding facts as a wrapper around out and run. Does it add something that I'm missing?

It only makes sense if there are generic assertions and receipt is just a variant of it, kind of like what sturct with known fields is to a map. If we say there is no generic assertions then it no longer matters.

In other words I was treating `about` as entity, and facts as set of `attribute → value` pairs for that entity. Trough this lens "receipt" is just a well defined relation akin of schema you can define (e.g. in datomic). My sentiment there is we could define other schemas for other relations over time. In fact someone could also extend receipts to add some extra attributes. e.g. I have contemplated idea of using this similar to session cookies.

Now non of that matters if we decide that generic assertions aren't a think we care in which case wrapper does become a more of redundancy.

Reply:
I think one other option to kind of do what I want in terms of generic assertions without having to burden anyone else is to change `about` to something that is clearly distinct from all other fields of the `args` in which case same goal could be accomplished. I have in fact considered following variant, but hesitated because of special relation IPLD has with `/`.

```js
{
   cmd: '/ucan/assert/receipt'
   args: {
       '/': 'bafy...thing',
       out: { ok: {} },
       run: []
   }
}
```

Some other distinct symbol could be utilized of course, but I don't know it never feels right, so I suggested `{about, facts}` instead.